pyiron_base/job/generic.py

```diff
@@ -500,7 +501,20 @@ def copy_to(self, project=None, new_job_name=None, input_only=False, new_databas
         if project is None and new_job_name is None:
             raise ValueError("copy_to requires either a new project or a new_job_name.")
 
-        in_same_project = project is None or project.path == self.project.path
+        if isinstance(project, self.project.__class__):
+            in_same_project = project.path == self.project.path
+            project = project or self.project
```

Comment:
Once we're inside one of the `isinstance(project, X)` clauses, what is the purpose of `or` in `project = project or self.project`? Won't it always evaluate to the first term?

If I understand that point correctly, then `project = project or self.project` can safely be extracted up out of the if clauses.

Reply:
Done